File: Python3/Hard/WrongAnswer/stamping-the-sequence_4.py
class Solution:
    def movesToStamp(self, stamp: str, target: str) -> List[int]:
        
        M, N = len(stamp), len(target)
        ans = []
        done = [False] * N
        def change(i, j, k):
            if all(done[j+k] or target[j+k] == stamp[k] for k in range(M)):
                for k in range(M):
                    done[j+k] = True
                ans.append(i)
                return True
            return False
        changed = True
        while changed:
            changed = False
            for i in range(N-M+1):
                if not done[i]:
                    changed |= change(i, i, 0)
            for i in range(N-M, -1, -1):
                if not done[i]:
                    changed |= change(i, i, 0)
        return ans[::-1] if all(done) else []

File: Java/Hard/Accepted/stamping-the-sequence_4.java
class Solution {
    public int[] movesToStamp(String stamp, String target) {
        
        int n = target.length();
        char[] t = target.toCharArray();
        boolean[] visited = new boolean[n];
        List<Integer> ans = new ArrayList<>();
        int count = 0;
        while (count < n) {
            boolean done = false;
            for (int i = 0; i <= n - stamp.length(); i++) {
                if (!visited[i] && canConvert(stamp, t, i)) {
                    ans.add(i);
                    count += convert(stamp, t, i);
                    visited[i] = true;
                    done = true;
                }
            }
            if (!done) {
                return new int[0];
            }
        }
        Collections.reverse(ans);
        int[] ret = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++) {
            ret[i] = ans.get(i);
        }
        return ret;
    }
    
    public boolean canConvert(String stamp, char[] t, int start) {
        for (int i = 0; i < stamp.length(); i++) {
            if (t[start + i] != '?' && t[start + i] != stamp.charAt(i)) {
                return false;
            }
        }
        return true;
    }
    
    public int convert(String stamp, char[] t, int start) {
        int count = 0;
        for (int i = 0; i < stamp.length(); i++) {
            if (t[start + i] != '?') {
                t[start + i] = '?';
                count++;
            }
        }
        return count;
    }
}